src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -27,5 +30,597 @@ protected static void ReportBadRefToken(TypeSyntax returnTypeSyntax, DiagnosticB
                 diagnostics.Add(ErrorCode.ERR_UnexpectedToken, refKeyword.GetLocation(), refKeyword.ToString());
             }
         }
+
+        protected virtual CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
+        {
+            return CustomAttributesBag<CSharpAttributeData>.Empty;
+        }
+
+        protected virtual CustomAttributesBag<CSharpAttributeData> GetReturnTypeAttributesBag()
+        {
+            return CustomAttributesBag<CSharpAttributeData>.Empty;
+        }
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations =>
+            DecodeReturnTypeAnnotationAttributes(GetDecodedReturnTypeWellKnownAttributeData());
+
+        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull
+            => GetDecodedReturnTypeWellKnownAttributeData()?.NotNullIfParameterNotNull ?? ImmutableHashSet<string>.Empty;
+
+        internal void GenerateExternalMethodWarnings(DiagnosticBag diagnostics)
+        {
+            if (this.GetAttributes().IsEmpty && !this.ContainingType.IsComImport)
+            {
+                // external method with no attributes
+                var errorCode = (this.MethodKind == MethodKind.Constructor || this.MethodKind == MethodKind.StaticConstructor) ?
+                    ErrorCode.WRN_ExternCtorNoImplementation :
+                    ErrorCode.WRN_ExternMethodNoImplementation;
+                diagnostics.Add(errorCode, this.Locations[0], this);
+            }
+        }
+
+#nullable enable
```

Comment:
Why not enable for this code block, basically including `GenerateExternalMethodWarnings`?